src/clj/ataru/virkailija/virkailija_routes.clj

```diff
@@ -98,8 +99,10 @@
                  (api/POST "/forms" {session :session}
                    :summary "Persist changed form."
                    :body [form ataru-schema/FormWithContent]
-                   (trying #(form-store/upsert-form
-                             (assoc form :modified-by (-> session :identity :username)))))
+                   (match (trying #(form-store/create-form-or-increment-version!
```

Comment:
Using `trying` (and the resulting "intermediary" compojure response) here was somewhat confusing at first sight, but seems to work :)